### ISSUE TITLE
Error constructing a ParameterFrame with defaults

### ISSUE THREAD

**Issue by hsaunders1904** [Fusion-Power-Plant-Framework/bluemira#1023]
## Describe the bug

When constructing a new `ParameterFrame`, if `with_defaults` is set to `True`, a `KeyError` is raised.

## Steps to reproduce

Steps to reproduce the behaviour:

```python
from bluemira.base.parameter import ParameterFrame

pf = ParameterFrame(with_defaults=True)
```

## Expected behaviour

No error. `ParameterFrame` constructed containing default values - which I guess in this case should be no values.

**Comment by hsaunders1904** [comment]
Fixing this seems easy, but testing it is not so simple. I can add a test like the following to `TestParameterFrame`. This passes when only `TestParameterFrame` is run, but fails when it's run as part of the test suite.

```python
class TestParameterFrame:

    def test_init_with_defaults_sets_no_parameters(self):
        pf = ParameterFrame(with_defaults=True)

        # Defaults are empty on base class; expect nothing to be set
        assert len(pf.get_parameter_list()) == 0
```

This is down to the class method `ParameterFrame.set_default_params`, which edits the `ParameterFrame` definition with some default parameters. The method is called in the class definition of `Configuration`. So, if the above test is run _before_ `Configuration` is imported somewhere, it passes. If it's run _after_ `Configuration` is imported, the test fails (this is easy to confirm by importing `Configuration` at the start of the test).

I think this could cause some subtle bugs down-the-line. Perhaps this issue can wait until we've had our discussions on `ParameterFrame`'s design.

**Comment by je-cook** [comment]
We talked in the past about removing the default mechanics from ParameterFrame and I think we ended on removing after we had removed BLUEPRINT and set up the new reactor build. Then we can move all the 'default' values out to some file. 
The default stuff has mutated to try and keep it about while that happens but there could well be problems that have been missed. I think waiting is best especially if we end up removing it
